sdk/core/Azure.Core/src/Shared/LruCache.cs

```diff
@@ -12,6 +14,7 @@ namespace Azure.Core
     /// <typeparam name="TKey">The type of key</typeparam>
     /// <typeparam name="TValue">The type of value</typeparam>
     internal class LruCache<TKey, TValue> : IEnumerable<KeyValuePair<TKey, TValue>>
+        where TKey : notnull
```

Comment:
Not a huge deal, but I think we can probably allow TValue to be null in the various methods/internal fields.